test/apps/iodemo/io_demo.cc

```diff
@@ -1410,6 +1409,42 @@ class DemoClient : public P2pDemoCommon {
         return data_size;
     }
 
+    static void
+    get_server_info_uncompleted_ops(const server_info_t& server_info,
+                                    std::stringstream &ss)
+    {
+        ss << get_num_uncompleted(server_info) << " uncompleted operations"
+           << " (read: " << server_info.num_completed[IO_READ] << "/"
+           << server_info.num_sent[IO_READ] << "; write: "
+           << server_info.num_completed[IO_WRITE] << "/"
+           << server_info.num_sent[IO_WRITE] << ")";
+    }
+
+    void get_uncompleted_servers_info(std::stringstream &ss)
+    {
+        size_t uncomp               = 0;
+        size_t uncomp_to_disconnect = 0;
+        size_t counter              = 1;
+
+        for (size_t i = 0; i < _active_servers.size(); ++i) {
+            if (get_num_uncompleted(_active_servers[i]) == 0) {
+                continue;
+            }
+
+            server_info_t& server_info = _server_info[_active_servers[i]];
+
+            ss << std::endl << counter++ << ". "
+               << server_info.conn->get_log_prefix() << " has ";
+            get_server_info_uncompleted_ops(server_info, ss);
+
+            uncomp++;
+            uncomp_to_disconnect += !server_info.conn->is_disconnecting();
```

Comment:
instead, maybe print for each connection if it's disconnecting inside get_server_info_uncompleted_ops
and rename get_server_info_uncompleted_ops to something like dump_server_info or server_info_serialize

Reply:
done